.gitignore

```diff
@@ -4,6 +4,8 @@
 .DS_Store
 .directory
 /WebKitBuild/
+/.clangd
+/update-compile-commands-symlink.conf
```

Comment:
As far I know, not really, as those ignore patterns would only apply inside `Tools/clangd`.

I tested doing the change just in case, and it indeed seem to only apply to `Tools/clangd` and not to the root of the repository, which sounds reasonable to me.